### ISSUE TITLE
Class:move should fail early when destination path is invalid

### ISSUE THREAD

**Issue by codedokode** [phpactor/phpactor#1804]
I thought that it is possible to specify a directory as a destination in `class:move` command and got an error after some of the files have been modified. I think that it makes sense to add some checks when destination is specified as a file path: 

- destination path doesn't exist
- destination path ends with `.php` (I think that both PSR-0 and PSR-4 require using `.php` suffix so there cannot be other extensions)

In any of these cases the command should fail early without making any changes. It also would be helpful to show a message that a directory cannot be used as a destination for move.

To reproduce, create following tree: 

```
├── composer.json
├── src
│   ├── A.php
│   ├── B
│   └── C.php
```
File contents: composer.json: 

```json
{
	"autoload": {
		"psr-4": {
			"TestNamespace\\": "src"
		}
	}
}
```
A.php: 

```php
<?php 

namespace TestNamespace;

class A {}
```

C.php: 

```php
<?php 

namespace TestNamespace;

use TestNamespace\A;

class C extends A {}
```

Run: 
```
composer dump-autoload
phpactor class:move -v src/A.php src/B/
```

Output:

```
[REPL] /tmp/test-move-bug/src/C.php
       38:53 TestNamespace\A => TestNamespace\B
       72:73 A => B
[REPL] /tmp/test-move-bug/src/A.php
       40:41 A => B
[MOVE] /tmp/test-move-bug/src/A.php => /tmp/test-move-bug/src/B/

In ClassMover.php line 88:
                                                                                     
  [RuntimeException]                                                                 
  Could not move file "/tmp/test-move-bug/src/A.php" to "/tmp/test-move-bug/src/B/"  

Exception trace:
  at /test/phpactor/lib/Extension/ClassMover/Application/ClassMover.php:88
 Phpactor\Extension\ClassMover\Application\ClassMover->moveFile() at /test/phpactor/lib/Extension/ClassMover/Application/ClassMover.php:66
 Phpactor\Extension\ClassMover\Application\ClassMover->move() at /test/phpactor/lib/Extension/ClassMover/Command/ClassMoveCommand.php:64
 Phpactor\Extension\ClassMover\Command\ClassMoveCommand->execute() at /test/phpactor/vendor/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at /test/phpactor/vendor/symfony/console/Application.php:1015
 Symfony\Component\Console\Application->doRunCommand() at /test/phpactor/vendor/symfony/console/Application.php:299
 Symfony\Component\Console\Application->doRun() at /test/phpactor/lib/Application.php:48
 Phpactor\Application->doRun() at /test/phpactor/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at /test/phpactor/bin/phpactor:47

In Filesystem.php line 294:
                                                                                
  [Symfony\Component\Filesystem\Exception\IOException]                          
  Cannot rename because the target "/tmp/test-move-bug/src/B/" already exists.  
                                                                                

Exception trace:
  at /test/phpactor/vendor/symfony/filesystem/Filesystem.php:294
 Symfony\Component\Filesystem\Filesystem->rename() at /test/phpactor/lib/Filesystem/Adapter/Simple/SimpleFilesystem.php:51
 Phpactor\Filesystem\Adapter\Simple\SimpleFilesystem->move() at /test/phpactor/lib/Extension/ClassMover/Application/ClassMover.php:114
 Phpactor\Extension\ClassMover\Application\ClassMover->doMoveFile() at /test/phpactor/lib/Extension/ClassMover/Application/ClassMover.php:86
 Phpactor\Extension\ClassMover\Application\ClassMover->moveFile() at /test/phpactor/lib/Extension/ClassMover/Application/ClassMover.php:66
 Phpactor\Extension\ClassMover\Application\ClassMover->move() at /test/phpactor/lib/Extension/ClassMover/Command/ClassMoveCommand.php:64
 Phpactor\Extension\ClassMover\Command\ClassMoveCommand->execute() at /test/phpactor/vendor/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at /test/phpactor/vendor/symfony/console/Application.php:1015
 Symfony\Component\Console\Application->doRunCommand() at /test/phpactor/vendor/symfony/console/Application.php:299
 Symfony\Component\Console\Application->doRun() at /test/phpactor/lib/Application.php:48
 Phpactor\Application->doRun() at /test/phpactor/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at /test/phpactor/bin/phpactor:47

```                                                                                     
As you see, the class references have been edited but the class itself wasn't moved.

**Comment by dantleech** [comment]
Feel free to open a PR, should be an easy pick :)

**Comment by dantleech** [comment]
Fixed in #1845
